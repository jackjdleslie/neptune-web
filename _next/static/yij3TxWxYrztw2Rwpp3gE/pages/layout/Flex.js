(window.webpackJsonp=window.webpackJsonp||[]).push([[74],{"41zC":function(n,e,t){(window.__NEXT_P=window.__NEXT_P||[]).push(["/layout/Flex",function(){var n=t("po96");return{page:n.default||n}}])},po96:function(n,e,t){"use strict";t.r(e),t.d(e,"meta",function(){return r}),t.d(e,"default",function(){return l});var o=t("pneb"),a=t("wk2l"),p=t("ERkP"),u=t.n(p),c=t("bBUn"),r=(u.a.createElement,{displayName:"Flex",isBeta:!0,size:"",slug:"Flex"}),i={meta:r},s="wrapper";function l(n){var e=n.components,t=Object(a.a)(n,["components"]);return Object(c.a)(s,Object(o.a)({},i,t,{components:e,mdxType:"MDXLayout"}),Object(c.a)("p",null,"TODO."))}l.isMDXComponent=!0}},[["41zC",1,0]]]);