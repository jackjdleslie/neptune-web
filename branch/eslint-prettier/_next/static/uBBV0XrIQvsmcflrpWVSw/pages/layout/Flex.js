(window.webpackJsonp=window.webpackJsonp||[]).push([[37],{"41zC":function(n,e,t){(window.__NEXT_P=window.__NEXT_P||[]).push(["/layout/Flex",function(){return t("po96")}])},po96:function(n,e,t){"use strict";t.r(e),t.d(e,"meta",function(){return i}),t.d(e,"default",function(){return w});var o=t("pneb"),a=t("wk2l"),p=t("ERkP"),u=t.n(p),c=t("bBUn"),i=(u.a.createElement,{displayName:"Flex",isBeta:!0,size:"",slug:"Flex"}),r={meta:i},s="wrapper";function w(n){var e=n.components,t=Object(a.a)(n,["components"]);return Object(c.a)(s,Object(o.a)({},r,t,{components:e,mdxType:"MDXLayout"}),Object(c.a)("p",null,"TODO."))}w.isMDXComponent=!0}},[["41zC",1,0]]]);