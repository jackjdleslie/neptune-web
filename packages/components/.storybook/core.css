:root {
  --primary-color: rgb(0, 185, 255);
  --pay-color: rgb(46, 208, 110);
  --danger-color: #f53636;
  --text-on-primary: white;

  --font: Averta, 'Avenir W02', Avenir, Helvetica, Arial, sans-serif;
  --weight-1: 400;
  --weight-2: 600;
  --weight-3: 800;

  --space-1: 8px;
  --space-2: 16px;
  --space-3: 24px;
  --space-4: 32px;
  --space-5: 40px;
  --space-6: 48px;

  --border-radius: 3px;
}

.primary-background {
  background: var(--primary-color);
}
