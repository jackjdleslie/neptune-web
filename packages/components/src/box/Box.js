/** @jsx jsx */
/* eslint-disable */
import { jsx } from '@emotion/core';
import Types from 'prop-types';

import { Size, JustifyContent, AlignItems, Breakpoint, Spacer } from '../common';

const mediaQueries = Object.values(Breakpoint).map(bp => `@media (min-width: ${bp}px)`);

const Box = props => {
  const {
    as: Element,
    size,
    justifyContent,
    alignItems,
    children,
    className,
    marginX,
    paddingX,
    marginY,
    paddingY,
    customMediaQueries,
  } = props;

  const getFlex = breakpoint => {
    const style = { flex: 0, display: 'none' };
    if (size && size[breakpoint]) {
      style.flex =
        size[breakpoint] <= 1 ? `0 1 ${size[breakpoint] * 100}%` : `0 0 ${size[breakpoint]}px`;
      style.display = 'flex';
    }
    return style;
  };

  const style = {
    [customMediaQueries[0]]: {
      ...getFlex(Size.EXTRA_SMALL),
    },
    [customMediaQueries[1]]: {
      ...getFlex(Size.SMALL),
    },
    [customMediaQueries[2]]: {
      ...getFlex(Size.MEDIUM),
    },
    [customMediaQueries[3]]: {
      ...getFlex(Size.LARGE),
    },
    [customMediaQueries[4]]: {
      ...getFlex(Size.EXTRA_LARGE),
    },
  };

  if (justifyContent) {
    style.justifyContent = justifyContent;
  }
  if (alignItems) {
    style.alignItems = alignItems;
  }

  style.margin = `${Spacer * marginY}px  ${Spacer * marginX}px`;
  style.padding = `${Spacer * paddingY}px  ${Spacer * paddingX}px`;

  return size ? (
    <Element className={className} css={style}>
      {children}
    </Element>
  ) : null;
};

Box.propTypes = {
  alignItems: Types.oneOf([...Object.values(AlignItems)]),
  children: Types.oneOfType([Types.arrayOf(Types.node), Types.node]),
  className: Types.string,
  justifyContent: Types.oneOf([...Object.values(JustifyContent)]),
  marginX: Types.number,
  marginY: Types.number,
  paddingX: Types.number,
  paddingY: Types.number,
  as: Types.elementType,
  size: Types.shape({
    lg: Types.number,
    md: Types.number,
    sm: Types.number,
    xl: Types.number,
    xs: Types.number,
  }).isRequired,
  customMediaQueries: Types.arrayOf(Types.string),
};

Box.defaultProps = {
  justifyContent: null,
  alignItems: null,
  children: null,
  className: '',
  paddingX: 0,
  marginX: 0,
  paddingY: 0,
  marginY: 0,
  as: 'div',
  customMediaQueries: mediaQueries,
};

export default Box;
