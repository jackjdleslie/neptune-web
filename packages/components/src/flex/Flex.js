/** @jsx jsx */
import React from 'react';
import { jsx } from '@emotion/core';
import Types from 'prop-types';
import { Size, FlexDirection, Breakpoint } from '../common';

const mediaQueries = Object.values(Breakpoint).map(bp => `@media (min-width: ${bp}px)`);

const Flex = props => {
  const {
    as: Element,
    direction,
    children,
    marginX,
    paddingX,
    marginY,
    paddingY,
    customMediaQueries,
    className,
  } = props;

  const getFlexDirection = breakpoint => {
    return direction && direction[breakpoint] && FlexDirection.indexOf(direction[breakpoint]) > -1
      ? direction[breakpoint]
      : 'row';
  };

  const style = {
    [customMediaQueries[0]]: {
      flexDirection: getFlexDirection(Size.EXTRA_SMALL),
    },
    [customMediaQueries[1]]: {
      flexDirection: getFlexDirection(Size.SMALL),
    },
    [customMediaQueries[2]]: {
      flexDirection: getFlexDirection(Size.MEDIUM),
    },
    [customMediaQueries[3]]: {
      flexDirection: getFlexDirection(Size.LARGE),
    },
    [customMediaQueries[4]]: {
      flexDirection: getFlexDirection(Size.EXTRA_LARGE),
    },
  };

  style.display = 'flex';

  return (
    <Element className={className} css={style}>
      {React.Children.map(children, child => {
        if (child && child.type && child.type.name === 'Box') {
          const childrenProps = { ...child.props, marginX, paddingX, marginY, paddingY };
          return React.cloneElement(child, childrenProps);
        }
        return child;
      })}
    </Element>
  );
};

Flex.propTypes = {
  direction: Types.shape({
    xs: Types.oneOf(FlexDirection),
    sm: Types.oneOf(FlexDirection),
    md: Types.oneOf(FlexDirection),
    lg: Types.oneOf(FlexDirection),
    xl: Types.oneOf(FlexDirection),
  }).isRequired,
  children: Types.oneOfType([Types.arrayOf(Types.node), Types.node]),
  marginX: Types.number,
  paddingX: Types.number,
  marginY: Types.number,
  paddingY: Types.number,
  customMediaQueries: Types.arrayOf(Types.string),
  as: Types.elementType,
  className: Types.string,
};

Flex.defaultProps = {
  as: 'div',
  children: null,
  marginX: 0,
  paddingX: 0,
  marginY: 0,
  paddingY: 0,
  customMediaQueries: mediaQueries,
  className: '',
};

export default Flex;
