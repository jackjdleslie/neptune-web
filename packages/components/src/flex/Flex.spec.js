import serializer from 'jest-emotion';
/** @jsx jsx */
import { jsx } from '@emotion/core';
import renderer from 'react-test-renderer'; // eslint-disable-line import/no-extraneous-dependencies

import Flex from './';
import Box from '../box';

const Direction = { xs: 'row', sm: 'row-reverse', md: 'column', lg: 'column-reverse', xl: 'row' };

const Size = { xs: 1 / 8, sm: 1 / 4, md: 1 / 2, lg: 100, xl: 100 };

const customMediaQueries = [0, 10, 20, 30, 40].map(bp => `@media (min-width: ${bp}px)`);

expect.addSnapshotSerializer(serializer);

const props = {
  direction: Direction,
  marginX: 0,
  marginY: 1,
  paddingX: 2,
  paddingY: 3,
};

describe('Box', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Flex {...props}>
        <Box size={Size}>This is a children.</Box>
      </Flex>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('renders correctly with custom media queries', () => {
    const component = renderer.create(
      <Flex {...props} customMediaQueries={customMediaQueries}>
        <Box size={Size}>This is a children.</Box>
      </Flex>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
