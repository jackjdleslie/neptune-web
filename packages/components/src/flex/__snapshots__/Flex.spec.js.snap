// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders correctly 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:0px) {
  .emotion-1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:576px) {
  .emotion-1 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }
}

@media (min-width:768px) {
  .emotion-1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:992px) {
  .emotion-1 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

@media (min-width:1200px) {
  .emotion-1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

.emotion-0 {
  margin: 8px 0px;
  padding: 24px 16px;
}

@media (min-width:0px) {
  .emotion-0 {
    -webkit-flex: 0 1 12.5%;
    -ms-flex: 0 1 12.5%;
    flex: 0 1 12.5%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:576px) {
  .emotion-0 {
    -webkit-flex: 0 1 25%;
    -ms-flex: 0 1 25%;
    flex: 0 1 25%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .emotion-0 {
    -webkit-flex: 0 1 50%;
    -ms-flex: 0 1 50%;
    flex: 0 1 50%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:992px) {
  .emotion-0 {
    -webkit-flex: 0 0 100px;
    -ms-flex: 0 0 100px;
    flex: 0 0 100px;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    -webkit-flex: 0 0 100px;
    -ms-flex: 0 0 100px;
    flex: 0 0 100px;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<section
  className="anotherClass1 anotherClass2 emotion-1"
>
  <div
    className=" emotion-0"
  >
    This is a children.
  </div>
</section>
`;

exports[`Box renders correctly with custom media queries 1`] = `
.emotion-0 {
  margin: 8px 0px;
  padding: 24px 16px;
}

@media (min-width:0px) {
  .emotion-0 {
    -webkit-flex: 0 1 12.5%;
    -ms-flex: 0 1 12.5%;
    flex: 0 1 12.5%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:576px) {
  .emotion-0 {
    -webkit-flex: 0 1 25%;
    -ms-flex: 0 1 25%;
    flex: 0 1 25%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .emotion-0 {
    -webkit-flex: 0 1 50%;
    -ms-flex: 0 1 50%;
    flex: 0 1 50%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:992px) {
  .emotion-0 {
    -webkit-flex: 0 0 100px;
    -ms-flex: 0 0 100px;
    flex: 0 0 100px;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    -webkit-flex: 0 0 100px;
    -ms-flex: 0 0 100px;
    flex: 0 0 100px;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:0px) {
  .emotion-1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:10px) {
  .emotion-1 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }
}

@media (min-width:20px) {
  .emotion-1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:30px) {
  .emotion-1 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

@media (min-width:40px) {
  .emotion-1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<section
  className="anotherClass1 anotherClass2 emotion-1"
>
  <div
    className=" emotion-0"
  >
    This is a children.
  </div>
</section>
`;
